### PR TITLE
Check quarkus.smallrye-openapi.store-schema-directory property and http Client improve coverage

### PR DESCRIPTION
**Pull request by pjgg**
OpenAPI and httpClient coverage

- Cover Open API store-schema-directory property.

- add some extra tests that cover GET/POTS/PUT/DELETE HTTP methods over sync/async(Mutiny) Rest API(Server and client-side).

### CODE REVIEW

File: 013-quarkus-oidc-restclient/pom.xml [hunk 1]
```diff
@@ -26,6 +30,15 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-rest-client</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-openapi</artifactId>
+        </dependency>
+<!--  Looks that it's not required, however Docs says the opposite. https://quarkus.io/guides/rest-client#async-support        -->
```
Comment (rsvoboda) [inline 013-quarkus-oidc-restclient/pom.xml:37]: I think it is thanks to `quarkus-resteasy-mutiny` dependency on line 17 + `quarkus-rest-client` on line 29.

the missing part is probably just `extensions/rest-client-mutiny/deployment/src/main/java/io/quarkus/restclient/mutiny/deployment/RestClientMutinyProcessor.java`

I think `quarkus-rest-client` could be removed and `quarkus-rest-client-mutiny` enabled to be safe for the future (if rest-client-mutiny gets enhanced)

Comment (rsvoboda) [reply 013-quarkus-oidc-restclient/pom.xml:37]: https://mvnrepository.com/artifact/io.quarkus/quarkus-rest-client-mutiny/1.11.0.Final

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/OpenApiTest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.smallrye.openapi.runtime.io.Format;
+
+@QuarkusTest
+public class OpenApiTest {
```
Comment (Sgitario) [inline 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/OpenApiTest.java:15]: Apart from the @rsvoboda 's comments, I'm missing the Native version of this test.

**Review by Sgitario — CHANGES_REQUESTED** [review]

File: 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/OpenApiTest.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    @Test
+    public void testOpenApiPathAccessResource() {
+        String jsonFileName = OPEN_API_DOT + Format.JSON.toString().toLowerCase();
+        Path json = Paths.get(directory, jsonFileName);
+        Assertions.assertTrue(Files.exists(json), directory + jsonFileName + " doesn't exist.");
+        String yamlFileName = OPEN_API_DOT + Format.YAML.toString().toLowerCase();
+        Path yaml = Paths.get(directory, yamlFileName);
+        Assertions.assertTrue(Files.exists(yaml), directory + yamlFileName +" doesn't exist.");
+    }
+}
```
Comment (rsvoboda) [inline 013-quarkus-oidc-restclient/src/test/java/io/quarkus/qe/OpenApiTest.java:29]: Please add test for content of generated json / yaml files.

**Review by Sgitario — APPROVED** [review]

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**